server/odc-service/src/main/java/com/oceanbase/odc/service/state/StateRouteAspect.java

```diff
@@ -0,0 +1,141 @@
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Preconditions;
+import com.oceanbase.odc.common.trace.TraceContextHolder;
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.common.util.SystemUtils;
+import com.oceanbase.odc.service.common.model.HostProperties;
+import com.oceanbase.odc.service.common.util.SpringContextUtil;
+import com.oceanbase.odc.service.dispatch.DispatchResponse;
+import com.oceanbase.odc.service.dispatch.RequestDispatcher;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Aspect
+@Slf4j
+@ConditionalOnProperty(value = {"odc.state.enabled"}, havingValue = "true")
+public class StateRouteAspect {
+
+    @Autowired
+    private HostProperties properties;
+
+    @Autowired
+    private RouteHealthManager routeHealthManager;
+    @Autowired
+    private RequestDispatcher requestDispatcher;
+
+    @Pointcut("@annotation(com.oceanbase.odc.service.state.StatefulRoute)")
+    public void stateRouteMethods() {}
+
+    @Around("stateRouteMethods()")
+    public Object aroundStateRouteExecution(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        MethodSignature signature = (MethodSignature) proceedingJoinPoint.getSignature();
+        Method method = signature.getMethod();
+        StatefulRoute statefulRoute = findAnnotation(method, StatefulRoute.class);
+        boolean nodeChanged = false;
+        StateManager stateManager = null;
+        RouteInfo routeInfo = null;
+        if (statefulRoute != null) {
+            Object stateIdBySePL = parseStateIdFromParameter(proceedingJoinPoint, statefulRoute.stateIdExpression());
+            stateManager = getStateManager(statefulRoute);
+            routeInfo = stateManager.getRouteInfo(stateIdBySePL);
+            if (!routeInfo.isCurrentNode(properties)) {
+                if (routeHealthManager.isHealthy(routeInfo)) {
+                    DispatchResponse dispatchResponse =
+                            requestDispatcher.forward(routeInfo.getHostName(), routeInfo.getPort());
+                    logTrace(method, stateIdBySePL, routeInfo);
+                    StateRouteFilter.getContext().setDispatchResponse(dispatchResponse);
+                    return null;
+                } else {
+                    nodeChanged = true;
```

Comment:
i did not got here, why it is nodeChanged while target host is not healty?

Reply:
If it has already been determined that the target node is not the current node, then forwarding should take place if the target node is healthy. If not  (which implies that the current node is needed to handle the request), then it means the node has changed.